### PR TITLE
Update host resource and Added CHAP Parameters

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
1. CHAP Parameters added
2. Update Resource added

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests

### CODE REVIEW

**Comment by Sakshi-dell** [comment]
File Coverage:
image

File: powerstore/resource_host.go [hunk 1]
```diff
@@ -92,6 +91,28 @@ func (r *resourceHost) Schema(ctx context.Context, req resource.SchemaRequest, r
...
+							Optional:            true,
+							Sensitive:           true,
+						},
+						"chap_mutual_username": schema.StringAttribute{
+							Description:         "Username for CHAP authentication. This value must be 1 to 64 UTF-8 characters. CHAP username is required when the cluster CHAP mode is mutual authentication.",
+							MarkdownDescription: "Username for CHAP authentication. This value must be 1 to 64 UTF-8 characters. CHAP username is required when the cluster CHAP mode is mutual authentication.",
+							Optional:            true,
+						},
+						"chap_single_password": schema.StringAttribute{
+							Description:         "Username for CHAP authentication. This value must be 1 to 64 UTF-8 characters. CHAP username is required when the cluster CHAP mode is single authentication.",
```
Comment (shenda1) [inline powerstore/resource_host.go:106]: Description is for username

File: powerstore/resource_host.go [hunk 2]
```diff
@@ -92,6 +91,28 @@ func (r *resourceHost) Schema(ctx context.Context, req resource.SchemaRequest, r
...
+							MarkdownDescription: "Username for CHAP authentication. This value must be 1 to 64 UTF-8 characters. CHAP username is required when the cluster CHAP mode is single authentication.",
+							Optional:            true,
+							Sensitive:           true,
+						},
+						"chap_single_username": schema.StringAttribute{
+							Description:         "Password for CHAP authentication. This value must be 12 to 64 UTF-8 characters. This password cannot be queried. CHAP password is required when the cluster CHAP mode is single authentication.",
```
Comment (shenda1) [inline powerstore/resource_host.go:111]: Description is for password

File: powerstore/resource_host.go [hunk 3]
```diff
@@ -92,6 +91,28 @@ func (r *resourceHost) Schema(ctx context.Context, req resource.SchemaRequest, r
 								string(gopowerstore.InitiatorProtocolTypeEnumFC),
 							)},
 						},
+						"chap_mutual_password": schema.StringAttribute{
+							Description:         "Password for CHAP authentication. This value must be 12 to 64 UTF-8 characters. This password cannot be queried. CHAP password is required when the cluster CHAP mode is mutual authentication.",
+							MarkdownDescription: "Password for CHAP authentication. This value must be 12 to 64 UTF-8 characters. This password cannot be queried. CHAP password is required when the cluster CHAP mode is mutual authentication.",
```
Comment (shenda1) [inline powerstore/resource_host.go:96]: Do we need validation for providing chap_mutual_username and chap_mutual_password together?

File: powerstore/resource_host.go [hunk 4]
```diff
@@ -92,6 +91,28 @@ func (r *resourceHost) Schema(ctx context.Context, req resource.SchemaRequest, r
...
+							Optional:            true,
+							Sensitive:           true,
+						},
+						"chap_mutual_username": schema.StringAttribute{
```
Comment (shenda1) [inline powerstore/resource_host.go:100]: Do we need validation for providing chap_single_username and chap_single_password together?

File: powerstore/resource_host.go [hunk 5]
```diff
@@ -277,14 +413,179 @@ func (r resourceHost) serverToState(plan, state *models.Host, response gopowerst
...
+	for _, initiator := range addInitiatorsMap {
+		portName := initiator.PortName.ValueString()
+		portType := gopowerstore.InitiatorProtocolTypeEnum(initiator.PortType.ValueString())
+		addInitiators = append(addInitiators, gopowerstore.InitiatorCreateModify{
+			PortName: &portName,
+			PortType: &portType,
+		})
+	}
+
+	var removeInitiators []string
```
Comment (shenda1) [inline powerstore/resource_host.go:528]: Add the comment for remove as well

File: powerstore/resource_host.go [hunk 6]
```diff
@@ -277,14 +413,179 @@ func (r resourceHost) serverToState(plan, state *models.Host, response gopowerst
...
+
+		initiators = append(initiators, initiatorModel)
 	}
 	state.Initiators = initiators
 	if operation == operationRead {
 		state.HostGroupID = types.StringValue(response.HostGroupID)
 	}
 }
+
+func (r resourceHost) planToServer(plan, state models.Host) *gopowerstore.HostModify {
```
Comment (AnikaAgiwal2711) [inline powerstore/resource_host.go:473]: Can we have proper comments above the function implementation ?

File: powerstore/resource_host.go (hunk 5 above)
Comment (Sakshi-dell) [reply powerstore/resource_host.go:528]: Added

File: powerstore/resource_host.go (hunk 6 above)
Comment (Sakshi-dell) [reply powerstore/resource_host.go:473]: Added

File: powerstore/resource_host.go (hunk 2 above)
Comment (Sakshi-dell) [reply powerstore/resource_host.go:111]: Updated

File: powerstore/resource_host.go (hunk 1 above)
Comment (Sakshi-dell) [reply powerstore/resource_host.go:106]: Updated

File: powerstore/resource_host.go (hunk 4 above)
Comment (Sakshi-dell) [reply powerstore/resource_host.go:100]: As discussed, validation is present from API side for length.

File: powerstore/resource_host.go (hunk 3 above)
Comment (Sakshi-dell) [reply powerstore/resource_host.go:96]: As discussed, validation is present from API side for length.

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**